widget/check_group.go

```diff
@@ -179,54 +185,87 @@ type checkGroupRenderer struct {
 	checks *CheckGroup
 }
 
+func (r *CheckGroup) countColumns() int {
+	if r.numColumns < 1 {
+		r.numColumns = 1
+	}
+	return r.numColumns
+}
+
+func (r *CheckGroup) countRows() int {
+	return int(math.Ceil(float64(len(r.items)) / float64(r.countColumns())))
+}
+
+func getLeading(size float64, offset int) float32 {
+	return float32(size * float64(offset))
+}
+
+func getTrailing(size float64, offset int) float32 {
+	return getLeading(size, offset+1)
+}
+
 // Layout the components of the checks widget
 func (r *checkGroupRenderer) Layout(_ fyne.Size) {
-	count := 1
-	if len(r.items) > 0 {
-		count = len(r.items)
-	}
-	var itemHeight, itemWidth float32
-	minSize := r.checks.MinSize()
+	cols := r.checks.countColumns()
+
+	primaryObjects := cols
+	secondaryObjects := r.checks.countRows()
 	if r.checks.Horizontal {
-		itemHeight = minSize.Height
-		itemWidth = minSize.Width / float32(count)
-	} else {
-		itemHeight = minSize.Height / float32(count)
-		itemWidth = minSize.Width
+		primaryObjects, secondaryObjects = secondaryObjects, primaryObjects
 	}
 
-	itemSize := fyne.NewSize(itemWidth, itemHeight)
-	x, y := float32(0), float32(0)
-	for _, item := range r.items {
-		item.Resize(itemSize)
-		item.Move(fyne.NewPos(x, y))
+	size := r.checks.MinSize()
+	cellWidth := float64(size.Width) / float64(primaryObjects)
```

Comment:
This doesn't seem quite right - normally a grid would spread out over the avilable space, but here it's using the minimum size instead. Is that intentional?

Initially it didn't matter because it was just 1 column but with horizontal and this grid layout I wonder if we should be respecting size...

Reply:
Lets leave it as the PR and if we want to change then review it separately - and probably match radio and anything else needing to also update.

Reply:
Fixed.